theseus/labs/lie/functional/se3_impl.py

```diff
@@ -21,85 +21,68 @@
 
 def check_group_tensor(tensor: torch.Tensor):
     def _impl(t_: torch.Tensor):
-        if t_.ndim != 3 or t_.shape[1:] != (3, 4):
-            raise ValueError(
-                f"SE3 data tensors can only be 3x4 matrices, but got shape {t_.shape}."
-            )
         SO3.check_group_tensor(t_[:, :, :3])
 
+    if tensor.shape[-2:] != (3, 4):
+        raise ValueError(
+            f"SE3 data tensors can only be 3x4 matrices, but got shape {tensor.shape}."
+        )
+
     checks_base(tensor, _impl)
 
 
 def check_matrix_tensor(tensor: torch.Tensor):
-    def _impl(t_):
-        if t_.ndim != 3 or t_.shape[-2:] != (3, 4):
-            raise ValueError(
-                f"SE3 data tensors can only be 3x4 matrices, but got shape {t_.shape}."
-            )
-
-    checks_base(tensor, _impl)
+    if tensor.shape[-2:] != (3, 4):
+        raise ValueError(
+            f"SE3 data tensors can only be 3x4 matrices, but got shape {tensor.shape}."
+        )
 
 
 def check_transform_tensor(tensor: torch.Tensor):
     SO3.check_transform_tensor(tensor)
 
 
 def check_tangent_vector(tangent_vector: torch.Tensor):
-    def _impl(t_: torch.Tensor):
-        _check = t_.ndim == 3 and t_.shape[1:] == (6, 1)
-        _check |= t_.ndim == 2 and t_.shape[1] == 6
-        if not _check:
-            raise ValueError(
-                f"Tangent vectors of SE3 should be 6-D vectors, "
-                f"but got shape {t_.shape}."
-            )
-
-    checks_base(tangent_vector, _impl)
+    _check = tangent_vector.shape[-2:] == (6, 1)
+    _check |= tangent_vector.shape[-1] == 6
+    if not _check:
+        raise ValueError(
+            f"Tangent vectors of SE3 should be 6-D vectors, "
+            f"but got shape {tangent_vector.shape}."
+        )
 
 
 def check_hat_matrix(matrix: torch.Tensor):
     def _impl(t_: torch.Tensor):
-        if t_.ndim != 3 or t_.shape[1:] != (4, 4):
-            raise ValueError("Hat matrices of SE(3) can only be 3x4 matrices")
-
-        if t_[:, -1].abs().max() > constants._SE3_NEAR_ZERO_EPS[t_.dtype]:
+        if matrix[:, -1].abs().max() > constants._SE3_NEAR_ZERO_EPS[matrix.dtype]:
```

Comment:
Shouldn't this have `matrix` -> `t_` and `:` -> `...`?

Reply:
Ah, good catch.